### PR TITLE
fix #7573 chore(nimbus): prefetch related entities for v6 api

### LINKED ISSUE #7573: Experimenter API failing during jetstream rerun
A `jetstream rerun --argo` command (when run in Argo on k8s) launches a pod for every day of analysis that needs to be conducted. These pods then proceed to each make a GET request to the Experimenter API ([https://experimenter.services.mozilla.com/api/v6/experiments/](https://experimenter.services.mozilla.com/api/v6/experiments/)). Recently, the API has started to return 502 errors in this situation. This does not _appear_ to be affecting the daily Jetstream jobs for in-flight experiments, it _appears_ to be localized to jetstream-config CI jobs and manually launched `jetstream rerun --argo` jobs.

┆Issue is synchronized with this [Jira Bug](https://mozilla-hub.atlassian.net/browse/EXP-2665)

### PR DESCRIPTION
**Pull request by jaredlockhart**
Because

* External callers depend on the V6 API for experiment metadata
* The V6 API was unoptimized for prefetching related entities and so was very slow

This commit

* Prefetches all related entities required for the V6 API
* Refactors methods to use the in memeory prefetched objects rather than creating additional database queries

### CODE REVIEW

**Review by yashika-khurana — APPROVED** [review]
That looks amazing 💪 👏 👏

File: app/experimenter/experiments/api/v6/serializers.py [hunk 1]
```diff
@@ -41,13 +41,15 @@ def get_feature(self, obj):
...
+            feature_config = sorted(feature_configs, key=lambda f: f.slug)[0]
             feature_config_slug = feature_config.slug
-            if obj.feature_values.filter(feature_config=feature_config).exists():
-                branch_feature_value = obj.feature_values.get(
-                    feature_config=feature_config
-                )
+            if feature_config in [fv.feature_config for fv in feature_values]:
+                branch_feature_value = [
+                    fv for fv in feature_values if fv.feature_config == feature_config
+                ][0]
```
Comment (eliserichards) [inline app/experimenter/experiments/api/v6/serializers.py:52]: As much as I hate that this has to have so much name repetition, you've made this really readable 😂 👏

File: app/experimenter/experiments/api/v6/serializers.py [hunk 2]
```diff
@@ -176,7 +178,9 @@ def get_branches(self, obj):
         return NimbusBranchSerializerSingleFeature(obj.branches.all(), many=True).data
 
     def get_featureIds(self, obj):
-        return list(obj.feature_configs.all().values_list("slug", flat=True))
+        return sorted(
```
Comment (eliserichards) [inline app/experimenter/experiments/api/v6/serializers.py:181]: `sorted()` 👏

File: app/experimenter/experiments/models.py [hunk 3]
```diff
@@ -410,14 +433,20 @@ def proposed_end_date(self):
...
+            c
+            for c in changes
+            if c.experiment_data["is_paused"]
+            and c.new_status == NimbusExperiment.Status.LIVE
+            and c.new_status_next is None
+            and c.new_publish_status == NimbusExperiment.PublishStatus.IDLE
+        ]
+
+        if paused_changelogs:
+            paused_change = sorted(paused_changelogs, key=lambda c: c.changed_on)[-1]
```
Comment (eliserichards) [inline app/experimenter/experiments/models.py:447]: what's the `-1` doing here? popping the changelog?

Comment (yashika-khurana) [reply app/experimenter/experiments/models.py:447]: so `sorted` returns the list, and [-1] meaning returning the last element of the list, so based on changes made, sorting based on changed on and returning results 🚀

**Pull request merged**